src/components/edit/thematic/ThematicDialog.js

```diff
@@ -259,40 +259,41 @@ export class ThematicDialog extends Component {
                                     errorText={indicatorError}
                                 />,
                             ]}
-                            {valueType === 'de' && [
-                                // Data element
-                                <DataElementGroupSelect
-                                    key="group"
-                                    dataElementGroup={dataElementGroup}
-                                    onChange={setDataElementGroup}
-                                    style={styles.select}
-                                />,
-                                dataElementGroup && (
-                                    <TotalsDetailsSelect
-                                        key="totals"
-                                        operand={operand}
-                                        onChange={setOperand}
-                                        style={styles.select}
-                                    />
-                                ),
-                                operand === true ? (
-                                    <DataElementOperandSelect
-                                        key="element"
+                            {valueType === 'de' ||
+                                (valueType === 'dc' && [
+                                    // Data element
+                                    <DataElementGroupSelect
+                                        key="group"
                                         dataElementGroup={dataElementGroup}
-                                        dataElement={dataItem}
-                                        onChange={setDataItem}
+                                        onChange={setDataElementGroup}
                                         style={styles.select}
-                                    />
-                                ) : (
-                                    <DataElementSelect
-                                        key="element"
-                                        dataElementGroup={dataElementGroup}
-                                        dataElement={dataItem}
-                                        onChange={setDataItem}
-                                        style={styles.select}
-                                    />
-                                ),
-                            ]}
+                                    />,
+                                    dataElementGroup && (
+                                        <TotalsDetailsSelect
+                                            key="totals"
+                                            operand={operand}
+                                            onChange={setOperand}
+                                            style={styles.select}
+                                        />
+                                    ),
+                                    operand === true || valueType === 'dc' ? (
```

Comment:
I believe `valueType` will ALWAYS `=== 'dc'` here since we're in the && condition of `valueType === 'dc'` above

Reply:
valueType can be 'de' or 'dc' - see comment above.